### DISCUSSION TITLE
Avoid vector DB to be updated each time when using RetrieveUserProxyAgent?

### DISCUSSION THREAD

**Discussion by bbence84** [microsoft/autogen#754] [Q&A]
Hi,
I am not even sure if this is what is happening, but it seems that each time I restart my script, the RAG agent recreates the embeddings. My retrieve_config:

```
        retrieve_config={
            "task": "qa",
            "collection_name": "docs",
            "docs_path": "rag_docs/",
            "custom_text_types": ["txt"],
            "chunk_token_size": 2000,
            "model": config_list[0]["model"],
            "client": chromadb.PersistentClient(path="./chroma"),
            "embedding_model": "all-mpnet-base-v2",
            "get_or_create": True,  # set to False if you don't want to reuse an existing collection, but you'll need to remove the collection manually
        }
```
I see the following in the console:

```
Trying to create collection.
2023-11-24 07:16:49 - Load pretrained SentenceTransformer: all-mpnet-base-v2
2023-11-24 07:16:51 - Found 12 chunks.
Batches: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:06<00:00,  6.37s/it] 
Batches: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 14.54it/s]
doc_ids:  [['doc_2', 'doc_8', 'doc_3']]

```

Does it mean that it only loads what is already there and does not recreate embeddings again?

If indeed it's just reusing it, then how can I force the update?

Thanks!
Bence

**Comment by sonichi** [comment]
@thinkall FYI

**Comment by thinkall** [comment]
Hi @bbence84 , if you set `get_or_create` to `True`, then it will just load the existing collection. If you want to recreate the indexes, you need to either use a new `collection_name` or remove the existing collection manually for now. We'll add a recreate option in #967 .

**Comment by tamish-max** [comment]
You are probably seeing **collection reuse plus retrieval**, not a full rebuild, because `get_or_create=True` with the same `PersistentClient(path="./chroma")` and `collection_name="docs"` should reopen the existing Chroma collection.

The confusing part is that the startup logs can still show document scanning / chunking-related messages even when the collection already exists.

Practical rules of thumb:
- same `client` path + same `collection_name` + `get_or_create=True` => reuse
- new `collection_name` => fresh collection
- deleting the collection (or the persisted Chroma data) => forced rebuild

If you want reliable updates when files change, I would version the collection by corpus hash / timestamp, or keep your own per-file fingerprints and rebuild only when content changes. The older AutoGen RAG examples generally do **not** do incremental invalidation for you.

One more thing to double-check: make sure `./chroma` resolves to the **same working directory** on every run. If your cwd changes, it can look like AutoGen is recreating embeddings when it is actually writing to a different persistence path.
